### ISSUE TITLE
php 8 #[UniqueConstraint] silently ignored without #[Table] attribute

### ISSUE THREAD

**Issue by Nemo64** [doctrine/doctrine2#8676]
I'm playing around with the doctrine 2.9 branch and attributes and notice that the syntax for indexes changed.
https://www.doctrine-project.org/projects/doctrine-orm/en/2.9/reference/attributes-reference.html#uniqueconstraint

```php
use Doctrine\ORM\Mapping\Entity;
use Doctrine\ORM\Mapping\UniqueConstraint;

#[Entity]
#[UniqueConstraint(name: "ean", columns=["ean"])]
class ECommerceProduct
{
}
```

I really like it, the nested annotations always looked ugly to me.
However, this example does not work. You must also add a `#[Table]` attribute, so this:

```php
use Doctrine\ORM\Mapping\Entity;
use Doctrine\ORM\Mapping\Table;
use Doctrine\ORM\Mapping\UniqueConstraint;

#[Entity]
#[Table]
#[UniqueConstraint(name: "ean", columns=["ean"])]
class ECommerceProduct
{
}
```

I guess that this makes sense, since the constraint mapping belongs to the table mapping, but the current behavior of silently ignoring isn't great. It should:
- either mark the schema as invalid
- or implicitly assume that you defined the table with all default definitions (which i'd prefer)

**Comment by beberlei** [comment]
Good catch! This is a bug

**Comment by beberlei** [comment]
Fixed in #8677
